src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -10,23 +10,23 @@ public interface IBitcoinSerializable
 
     public static class BitcoinSerializableExtensions
     {
-        public static void ReadWrite(this IBitcoinSerializable serializable, Stream stream, bool serializing, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null)
+        public static void ReadWrite(this IBitcoinSerializable serializable, Stream stream, bool serializing, ConsensusFactory consensusFactory, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
```

Comment:
I think maybe perhaps we should create a  default `ConsensusFactory` that will just return null (or throw when one of the types it expect is asked) for every type.
then this method can be called with no `ConsensusFactory` and still work.

Reply:
Hmmm I'm really not a fan of these methods that can be called with null defaults. I think in the solution this is something that we should be moving away from. I would much rather create another `ReadWrite` where we don't pass in a `ConsensusFactory`, should the need arise. This way we are more explicit as to the intent of the method.

Reply:
But there are many tyoes that use the serializer apart for the (Header,Block,Transaction) those dont need the factory.

Reply:
Yes I understand that :) but I think that the part you are referring to is here:

https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/NBitcoin/BitcoinStream.cs#L194

So here, if we are serializing/deserializing an object that is NOT Header,Block,Transaction we will just use the Activator?

Why would all the tests pass and this branch be able to sync if this was an issue? (Just asking)

:)